test/gui/shared/steps/steps.py

```diff
@@ -542,12 +435,9 @@ def step(context, receiver, resource):
 
 @Then('the error text "|any|" should be displayed in the sharing dialog')
 def step(context, fileShareContext):
+    shareItem = SharingDialog()
     test.compare(
-        str(
-            waitForObjectExists(
-                names.sharingDialog_The_file_can_not_be_shared_because_it_was_shared_without_sharing_permission_QLabel
-            ).text
-        ),
+        str(waitForObjectExists(shareItem.ERROR_SHOWN_ON_SHARING_DIALOG).text),
```

Comment:
is there a way to get if the object is visible or not on the page in squish. If that is possible you can make a method that returns true/false whether an element is visible or not and check the assertion based on that.
since we are using page objects now I think we should avoid calling the squish API on the page (such as `waitForObjectExists` from the step definitions)

Reply:
I think `waitForObjectExists` is the only way to do this.

Reply:
there is a function `findObject` which throws LookupError if object is not found.  you can use try, except in the function,